sphinx/screens/chats/chat-common/chat-common/src/main/java/chat/sphinx/chat_common/ui/ChatFragment.kt

```diff
@@ -711,6 +751,58 @@ abstract class ChatFragment<
             }
         }
 
+        onStopSupervisor.scope.launch(viewModel.mainImmediate) {
+            viewModel.getAttachmentFullscreenViewStateFlow().collect { viewState ->
+                attachmentFullscreenBinding.apply {
+                    @Exhaustive
+                    when (viewState) {
+                        is AttachmentFullscreenViewState.Idle -> {
+                            root.gone
+                            imageViewAttachmentFullscreen.setImageDrawable(null)
+                        }
+                        is AttachmentFullscreenViewState.Fullscreen -> {
+                            imageViewAttachmentFullscreen.scaleX = 1.0f
+                            imageViewAttachmentFullscreen.scaleY = 1.0f
+                            imageViewAttachmentFullscreen.translationX = 0f
+                            imageViewAttachmentFullscreen.translationY = 0f
+
+                            if (viewState.media?.localFile != null) {
+                                imageLoader.load(
+                                    imageViewAttachmentFullscreen,
+                                    viewState.media.localFile!!
+                                )
```

Comment:
this should be launched from a new coroutine + scope and have the job + disposable tracked such that upon ViewState change it can be cancelled.